polaris-react/src/components/SearchableListbox/SearchableListbox.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {useState, ReactNode, ReactElement} from 'react';
+
+import {Popover} from '../Popover';
+import {Scrollable} from '../Scrollable';
+import {AutoSelection, Listbox} from '../Listbox';
+import {useUniqueId} from '../../utilities/unique-id';
+
+import {StopPropagation, Search, SearchEmptyState} from './components';
+import styles from './SearchableListbox.scss';
+
+export interface SearchableListItem {
+  children: string | ReactNode;
+  value: string;
+  selected?: boolean;
+}
+
+export interface SearchableListboxProps {
+  activatorNode: ReactElement;
+  open: boolean;
+  showSearch: boolean;
+  searchValue: string;
+  searchEmptyStateMessage: string;
+  searchLabel: string;
+  searchPlaceholder?: string;
+  listItems?: SearchableListItem[];
+  loading?: boolean;
+  loadingAccessibilityLabel?: string;
+  footerAction?: ReactElement;
+  onClose(): void;
+  onSearch(search: string): void;
```

Comment:
While trying out the component, I noticed that it felt a bit odd to have `onSearch`, `searchLabel, `searchValue` and `showSearch` all be required props on the component. Since I wanted to try an example without the search box, I needed to explicitly set `showSearch={false}`, and create a few unused defaults for the other related props. As long as it doesn't cause other issues, it might be nice for this component to have default values for them instead :)